authors/apps/articles/tests/test_rate_article.py

```diff
@@ -0,0 +1,170 @@
+        author = self.user
+        article = Article(
+            author=author,
+            slug=self.slug,
+            body=body,
+            title=title,
+            description=description)
+        article.save()
+        self.rate_details = {
+            "user": {
+                "slug": self.slug,
+                "rate": 3
+            }
+        }
+        self.rate_url = "http://localhost:8000/api/article/"\
+            + self.slug + "/rate/"
+        self.view_rates_url = "http://localhost:8000/api/article/rate/"
+
+    def test_rate_article_without_token(self):
+        """
+        test weather rate article without token will fail.
+        """
+        response = self.client.post(
+            self.rate_url,
+            self.rate_details,
+            format='json')
+        self.assertIn(
+            'Authentication credentials were not provided.', str(
+                response.data))
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_rate_article(self):
+        """
+        test rate article.
+        """
+        self.client.credentials(HTTP_AUTHORIZATION='Bearer ' + self.token)
+        response = self.client.post(
+            self.rate_url,
+            self.rate_details,
+            format='json')
+        self.assertIn(
+            'sucessfully rated', str(
+                response.data))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_rate_article_not_found(self):
+        """
+        test weather rate article without article will fail.
+        """
+        self.client.credentials(HTTP_AUTHORIZATION='Bearer ' + self.token)
+        self.rate_details["user"]["slug"] = "-ss-dd-dd-ff"
+        response = self.client.post(
+            self.rate_url,
+            self.rate_details,
+            format='json')
+        self.assertIn(
+            'Article not found', str(
+                response.data))
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_rate_article_invalid_rate(self):
+        """
+        test rate article with invalid data wil fail.
+        """
+        self.client.credentials(HTTP_AUTHORIZATION='Bearer ' + self.token)
+        self.rate_details["user"]["rate"] = 7
+        response = self.client.post(
+            self.rate_url,
+            self.rate_details,
+            format='json')
+        self.rate_details["user"]["rate"] = 0
+        resp = self.client.post(
+            self.rate_url,
+            self.rate_details,
+            format='json')
+        self.assertIn(
+            'invalid rate value should be > 0 or <=5', str(
+                response.data))
+        self.assertIn(
+            'invalid rate value should be > 0 or <=5', str(
+                resp.data))
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_get_rate_article(self):
+        """
+        test weather rate article with token.
```

Comment:
typo here: `test whether...`
Also provide more context in your docstring because even with the typo fixed, the statement doesn't make sense. `test whether rate article with token.`